xds/internal/resolver/xds_resolver.go

```diff
@@ -0,0 +1,103 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package resolver implements the xds resolver.
+//
+// At this point, the resolver is named xds_experimental, and doesn't do very
+// much at all, except for returning a hard-coded service config which selects
+// the xds_experimental balancer.
+package resolver
+
+import (
+	"fmt"
+	"sync"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/internal"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+const (
+	xdsExperimental = "xds_experimental"
+
+	// The JSON form of the hard-coded service config which picks the
+	// xds_experimental balancer with round_robin as the child policy.
+	jsonSC = `{
+    "loadBalancingConfig":[
+      {
+        "xds_experimental":{
+          "childPolicy":[
+            {
+              "round_robin": {}
+            }
+          ]
+        }
+      }
+    ]
+  }`
+)
+
+var (
+	parseOnce sync.Once
+	parsedSC  serviceconfig.Config
+)
+
+// NewBuilder creates a new implementation of the resolver.Builder interface
+// for the xDS resolver.
+func NewBuilder() resolver.Builder {
+	parseOnce.Do(func() {
+		// The xds balancer must have been registered at this point for the service
+		// config to be parsed properly.
+		psc, err := internal.ParseServiceConfig(jsonSC)
```

Comment:
Actually, you have to move this to `Build` after #2951 because `internal.ParseServiceConfig()` will become `cc.ParseServiceConfig()`...

Reply:
Hmm ... I've left it as is for now. Whichever one goes in second has to resolve the conflict I guess :)